File: csharp/difference-of-squares/DifferenceOfSquares.cs
using System;

public static class DifferenceOfSquares
{
    public static int CalculateSquareOfSum(int max)
    {
        int sum = 0;
        for (int i = 1; i <= max; i++)
        {
            sum += i;
        }

        return (int) Math.Pow(sum, 2);
    }

    public static int CalculateSumOfSquares(int max)
    {
        int sum = 0;
        
        for (int i = 1; i <= max; i++)
        {
            sum += (int) Math.Pow(i, 2);
            
        }
        return sum;
    }

    public static int CalculateDifferenceOfSquares(int max)
    {
        return CalculateSquareOfSum(max) - CalculateSumOfSquares(max);
    }
}

File: vbnet/difference-of-squares/DifferenceOfSquares.vb
Imports System

Public Module DifferenceOfSquares
    Public Function CalculateSquareOfSum(ByVal max As Integer) As Integer
        Return CInt(Math.Pow(Enumerable.Range(1, max).Aggregate(Function(acc, n) acc + n), 2))
    End Function

    Public Function CalculateSumOfSquares(ByVal max As Integer) As Integer
        Return CInt(Enumerable.Range(1, max).Select(Function(n) Math.Pow(n, 2)).Aggregate(Function(acc, n) acc + n))
    End Function

    Public Function CalculateDifferenceOfSquares(ByVal max As Integer) As Integer
        Return CalculateSquareOfSum(max) - CalculateSumOfSquares(max)
    End Function
End Module
